proto/waste.proto

```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
+
+package smartcore.bos;
+
+option go_package = "github.com/vanti-dev/sc-bos/pkg/gen";
+
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "types/info.proto";
+
+// WasteApi exposes details about units of waste that have been produced by the building
+service WasteApi {
+  rpc ListWasteRecords(ListWasteRecordsRequest) returns (ListWasteRecordsResponse);
+  rpc PullWasteRecords(PullWasteRecordsRequest) returns (stream PullWasteRecordsResponse);
+}
+
+service WasteInfo {
+  rpc DescribeWasteRecord(DescribeWasteRecordRequest) returns (WasteRecordSupport);
+}
+
+message WasteRecord {
+  // The id of the waste record assigned by the external waste management system
+  string id = 1;
+  // The time the record was created, this is not the time the waste was created
+  google.protobuf.Timestamp recordCreatedTime = 2;
+  // The weight of the waste. Use the InfoApi to find the actual unit the device uses.
+  float weight = 3;
+  // The system the waste was collected from
+  string system = 4;
+  // True if the waste was recycled
+  bool recycled = 5;
+  // Area the waste was collected from
+  string area = 6;
+  // The date/time the waste was created
+  google.protobuf.Timestamp wasteCreatedTime = 7;
+  // The stream the waste was collected from. For example, general waste, mixed recycling etc.
+  string stream = 8;
+}
+
+message ListWasteRecordsResponse {
+  repeated WasteRecord wasteRecords = 1;
+
+  // A token, which can be sent as `page_token` to retrieve the next page.
+  // If this field is omitted, there are no subsequent pages.
+  string next_page_token = 2;
+  // If non-zero this is the total number of alerts after filtering is applied.
+  // This may be an estimate.
+  int32 total_size = 3;
+}
+
+message ListWasteRecordsRequest {
+  string name = 1;
+  google.protobuf.FieldMask read_mask = 2;
+}
+
+message PullWasteRecordsRequest {
+  string name = 1;
+  google.protobuf.FieldMask read_mask = 2;
+  bool updates_only = 3;
+}
+
+message PullWasteRecordsResponse {
+  repeated Change changes = 1;
+
+  message Change {
+    string name = 1;
+    google.protobuf.Timestamp change_time = 2;
+    WasteRecord waste_record = 3;
```

Comment:
For collection resources (those with List apis) the pull.change should look more like `{old_value, new_value, change_type}`